src/code_ocean_api/code_ocean_api.py

```diff
@@ -0,0 +1,249 @@
+"""Module to interface with codeocean api.
+"""
+import requests
+from typing import List, Dict
+
+class CodeOceanDataAssetRequests():
+    """This class will handle the methods needed to manage data assets stored
+    on Code Ocean's platform.
+    """
+
+    def __init__(self, domain:str, token:str) -> None:
+        """
+        Parameters
+        ---------------
+        domain : string
+            Domain of VPC
+        token : string
+            Authorization token
+        """
+
+        self.domain = domain
+        self.token = token
+        self.api_version = 1
+        self.asset_url = f"/api/v{self.api_version}/data_assets"
+
+    def get_data_asset(self, data_asset_id:str) -> dict:
+        """
+        This will get data from a GET request to code ocean API.
+        
+        Parameters
+        ---------------
+        data_asset_id : string
+            ID of the data asset
+        
+        Returns
+        ---------------
+        A json object as documented by CodeOcean's v1 api docs.
+        """
+
+        url = f"{self.domain}{self.asset_url}/{data_asset_id}"
+        response = requests.get(url, auth=(self.token, ""))
+        return response.json()
+
+    def register_data_asset(self, 
+        asset_name:str,
+        mount:str,
+        bucket:str,
+        prefix:str,
+        access_key_id:str,
```

Comment:
do you want to handle AWS credentials as an argument or via `self` like you did with the codocean token?

Reply:
I think for now we can pass it via a string. We may want to use botocore down the road. This api matches CodeOcean's REST api as much as possible. I think it might be better to make it as generic as possible so it can be open sourced more easily and maybe we can get contributions from other code ocean users.

Reply:
okay by me